xbmc/video/ContextMenus.cpp

```diff
@@ -168,4 +171,26 @@ bool CPlay::Execute(const CFileItemPtr& itemIn) const
   return true;
 };
 
+
+std::string CFavorite::GetLabel(const CFileItem& item) const
+{
+  //The 0 we're passing for contextWindow should only matter to folders
+  return g_localizeStrings.Get(XFILE::CFavouritesDirectory::IsFavourite(std::make_shared<CFileItem>(item).get(), 0) ? 14077 : 14076); // Add/Remove Favourite
```

Comment:
Hold please, we should definitely *not* do this. This can be executed from anywhere, and you can't assume one of those windows are present. On homescreen this will not give you the video window.

Actually, the entire point of the context menus redesign in the first place, was to get rid of exactly this kind of global states and window dependencies that caused them all to break when used from elsewhere.